sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -696,6 +696,37 @@ case class ResolvedStar(expressions: Seq[NamedExpression]) extends Star with Une
   override def toString: String = expressions.mkString("ResolvedStar(", ", ", ")")
 }
 
+/**
+ * Represents all input attributes to a given relational operator.
+ * This is used in Spark Connect dataframe, for example:
+ *    df1 = spark.createDataFrame([{"id": 1}])
+ *    df2 = spark.createDataFrame([{"id": 1, "val": "v"}])
+ *    df1.join(df2, "id").select(df1["*"])
+ * @param planId the plan id of target node.
+ */
+case class UnresolvedDataFrameStar(planId: Long) extends Star with Unevaluable {
```

Comment:
This does not need to be a star. It's just a placeholder and will be replaced by `ResolvedStar` after finding the matching plan.

Reply:
Then we can handle it in `ColumnResolutionHelper` and reuse code.

Reply:
got it